src/index.ts

```diff
@@ -1,49 +1,115 @@
 import { Bee } from '@ethersphere/bee-js'
 import { AsyncQueue, Binary, Chunk, MerkleTree } from 'cafe-utility'
 import { existsSync, readFileSync, writeFileSync } from 'fs'
-import { createServer } from 'http'
+import { createServer, Server } from 'http'
 import createKeccakHash from 'keccak'
 import { Stamper } from './stamper'
+import { getConfig } from './config'
+import { log, error } from './logger'
 
-const privateKey = 0x1234567812345678123456781234567812345678123456781234567812345678n
-const batchIdString = '0x1234567812345678123456781234567812345678123456781234567812345678'
-const depth = 17
-const target = 'http://localhost:1633'
+process.on('uncaughtException', (err: unknown) => {
+    error({ err }, 'Uncaught Exception');
+});
+
+process.on('unhandledRejection', (reason: unknown) => {
+    error({ err: reason }, 'Unhandled Rejection');
+});
 
 Chunk.hashFunction = (data: Uint8Array): Uint8Array => {
     return createKeccakHash('keccak256').update(Buffer.from(data)).digest()
 }
 
-const bee = new Bee(target)
-const batchId = Binary.hexToUint8Array(batchIdString)
-const path = `${batchIdString}.bin`
-
-const stamper = existsSync(path)
-    ? Stamper.fromState(privateKey, batchId, new Uint32Array(readFileSync(path)), depth)
-    : Stamper.fromBlank(privateKey, batchId, depth)
-
-let stampings = 0
-
-const server = createServer((request, response) => {
-    const before = Date.now()
-    const chunks: Buffer[] = []
-    request.on('data', chunk => chunks.push(chunk))
-    request.on('end', async () => {
-        const queue = new AsyncQueue(64, 64)
-        const data = Buffer.concat(chunks)
-        const tree = new MerkleTree(async chunk => {
-            await queue.enqueue(async () => {
-                const envelope = stamper.stamp(chunk)
-                await bee.uploadChunk(envelope, chunk.build())
-                stampings++
-            })
+(async () => {
+    const { beeEndpoint, batchId: batchIdString, port, privateKey, stamperPath } = await getConfig();
+    log({ beeEndpoint, port, batchId: batchIdString }, 'Configuration');
+    const bee = new Bee(beeEndpoint);
+    const { depth } = await bee.getTopology()
```

Comment:
@xFJA the batch depth value we need can be found using  `getGlobalPostageBatches` and filtering results by batchId.